src/mono/netcore/sample/iOS/HelloiOS/runtime.m

```diff
@@ -0,0 +1,377 @@
+{
+    mono_aot_register_module (mono_aot_module_Program_info);
+    mono_aot_register_module (mono_aot_module_System_Private_CoreLib_info);
+    mono_aot_register_module (mono_aot_module_System_Runtime_info);
+    mono_aot_register_module (mono_aot_module_System_Runtime_Extensions_info);
+    mono_aot_register_module (mono_aot_module_System_Collections_info);
+    mono_aot_register_module (mono_aot_module_System_Core_info);
+    mono_aot_register_module (mono_aot_module_System_Threading_info);
+    mono_aot_register_module (mono_aot_module_System_Threading_Tasks_info);
+    mono_aot_register_module (mono_aot_module_System_Linq_info);
+    mono_aot_register_module (mono_aot_module_System_Memory_info);
+    mono_aot_register_module (mono_aot_module_System_Runtime_InteropServices_info);
+    mono_aot_register_module (mono_aot_module_System_Text_Encoding_Extensions_info);
+    mono_aot_register_module (mono_aot_module_Microsoft_Win32_Primitives_info);
+    mono_aot_register_module (mono_aot_module_System_Console_info);
+    mono_aot_register_module (mono_aot_module_Program_info);
+}
+
+void mono_ios_setup_execution_mode (void)
+{
+    mono_jit_set_aot_mode (MONO_AOT_MODE_FULL);
+}
+
+void
+mono_ios_runtime_init (void)
+{
+    NSBundle *main_bundle = [NSBundle mainBundle];
+    NSString *path;
+    char *result;
+    int res, nargs, config_nargs;
+    char *executable;
+
+    // for now, only Invariant Mode is supported (FIXME: integrate ICU)
+    setenv ("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", "1", TRUE);
+
+    stdout_log = os_log_create ("com.dotnet.mono", "stdout");
+
+    bool wait_for_debugger = FALSE;
+    executable = "Program.dll";
+
+    const char *bundle = get_bundle_path ();
+    chdir (bundle);
+
+    register_dllmap ();
+
+    // register modules
+    mono_ios_register_modules ();
+    mono_ios_setup_execution_mode ();
+    
+    mono_debug_init (MONO_DEBUG_FORMAT_MONO);
+    mono_install_assembly_preload_hook (assembly_preload_hook, NULL);
+    mono_install_load_aot_data_hook (load_aot_data, free_aot_data, NULL);
+    mono_install_unhandled_exception_hook (unhandled_exception_handler, NULL);
+    mono_trace_init ();
+    mono_trace_set_log_handler (log_callback, NULL);
+    mono_set_signal_chaining (TRUE);
+    mono_set_crash_chaining (TRUE);
+
+    if (wait_for_debugger) {
+        char* options[] = { "--debugger-agent=transport=dt_socket,server=y,address=0.0.0.0:55555" };
+        mono_jit_parse_options (1, options);
+    }
+    mono_jit_init_version ("Mono.ios", "mobile");
+    mono_gc_init_finalizer_thread ();
+
+    MonoAssembly *assembly = load_assembly (executable, NULL);
+    assert (assembly);
+    os_log_info (OS_LOG_DEFAULT, "Executable: %{public}s", executable);
+
+    res = mono_jit_exec (mono_domain_get (), assembly, 1, &executable);
+    // Print this so apps parsing logs can detect when we exited
+    os_log_info (OS_LOG_DEFAULT, "Exit code: %d.", res);
+}
+
+// ICALLS used by the mobile profile of corelib
```

Comment:
None of the bellows should be needed. Please remove the dependency from mono instead

Reply:
@marek-safar `xamarin_log` is needed to properly implement `System.Console` (I have a draft for ConsolePal.iOS.cs).
`xamarin_GetFolderPath` most likely will be needed for Environment.GetFolderPath for iOS. `xamarin_timezone_get_data `, `xamarin_timezone_get_local_name` and `xamarin_timezone_get_names` will be needed to implement a portable small footprint ICU without 3rd parties.

But yeah let me delete them all and add back only when absolutely necessary .

Reply:
xamarin_log might be needed for hello world before we have the implementation in BCL but it should be simplified (this is hello world). The rest is not and will be handled most likely differently anyway